pkg/kustomize/secrets_test.go

```diff
@@ -319,94 +317,146 @@ func TestContainsComponentConfig(t *testing.T) {
-		assert := assert.New(t)
-		// Empty certs, should generate own
-		recPublicKey, recPrivateKey, err := EnsureTLSFor(quayContextWithBuilder, quayRegistry, emptyCert, emptyCert)
-		if err != nil {
-			t.Errorf(err.Error())
-		}
-		assert.NotEqual(recPublicKey, emptyCert)
-		assert.NotEqual(recPrivateKey, emptyCert)
-	})
+func TestEnsureTLSFor(t *testing.T) {
+	assert := assert.New(t)
 
-	t.Run("Quay context has buildman hostname. Certs do not contain buildman hostname. Operator should generate own.", func(t *testing.T) {
-		assert := assert.New(t)
-		// Buildman missing from certs, should generate own
-		recPublicKey, recPrivateKey, err := EnsureTLSFor(quayContextWithBuilder, quayRegistry, pubWithoutBuilder, privWithoutBuilder)
-		if err != nil {
-			t.Errorf(err.Error())
-		}
-		assert.NotEqual(recPublicKey, pubWithoutBuilder)
-		assert.NotEqual(recPrivateKey, privWithoutBuilder)
-	})
+	for _, test := range ensureTLSForTests {
+		quayRegistry := quayRegistry("test")
 
-	t.Run("Quay context has buildman hostname. Certs contain buildman hostname. Operator should not generate own.", func(t *testing.T) {
-		assert := assert.New(t)
-		// Buildman present in certs, should not generate
-		recPublicKey, recPrivateKey, err := EnsureTLSFor(quayContextWithBuilder, quayRegistry, pubWithBuilder, privWithBuilder)
-		if err != nil {
-			t.Errorf(err.Error())
+		quayContext := quaycontext.QuayRegistryContext{
+			ServerHostname:       test.serverHostname,
+			BuildManagerHostname: test.buildManagerHostname,
 		}
-		assert.Equal(recPublicKey, pubWithBuilder)
-		assert.Equal(recPrivateKey, privWithBuilder)
-	})
 
-	t.Run("Quay context does not have buildman hostname. Certs do not contain buildman hostname. Operator should not generate own.", func(t *testing.T) {
-		assert := assert.New(t)
-		// Buildman not in certs, not in context, should not generate
-		recPublicKey, recPrivateKey, err := EnsureTLSFor(quayContextWithoutBuilder, quayRegistry, pubWithoutBuilder, privWithoutBuilder)
-		if err != nil {
-			t.Errorf(err.Error())
-		}
-		assert.Equal(recPublicKey, pubWithoutBuilder)
-		assert.Equal(recPrivateKey, privWithoutBuilder)
-	})
+		tlsCert, tlsKey, err := EnsureTLSFor(&quayContext, quayRegistry, test.providedCert, test.providedKey)
 
-	t.Run("Quay context does not have buildman hostname. Certs contain buildman hostname. Operator should not generate own.", func(t *testing.T) {
-		assert := assert.New(t)
-		// Buildman not in certs, not in context, should not generate
-		recPublicKey, recPrivateKey, err := EnsureTLSFor(quayContextWithoutBuilder, quayRegistry, pubWithBuilder, privWithBuilder)
-		if err != nil {
-			t.Errorf(err.Error())
-		}
-		assert.Equal(recPublicKey, pubWithBuilder)
-		assert.Equal(recPrivateKey, privWithBuilder)
-	})
+		assert.Equal(test.expectedErr, err, test.name)
 
+		if test.providedCert == nil || test.providedKey == nil {
+			shared.ValidateCertPairWithHostname(tlsCert, tlsKey, test.serverHostname, fieldGroupNameFor("rout"))
```

Comment:
Is this supposed to be "rout" and not "route"?

Reply:
Yep